### PR TITLE
feat: redirect support `options.duplicate`

### PR DESCRIPTION
**Pull request by atian25**
most the time, when we custom a logger, want to:
- view all error log at one place, such as `common-error.log`
- but we also need to view the full log at the custom log file

so, change `redirect(level, logger)` to `redirect(level, logger, { duplicate: true })` to support this feature.

### CODE REVIEW

**Comment by atian25** [comment]
cc @dead-horse 这个也看看，我想把 egg-schedule 的日志改为双写的。

**Comment by atian25** [comment]
修改了，再看看

**Comment by dead-horse** [comment]
2.1.0

File: test/lib/logger.test.js [hunk 1]
```diff
@@ -126,18 +130,20 @@ describe('test/lib/logger.test.js', () => {
       level: levels.INFO,
     }));
     logger1.redirect('warn', logger2);
-    logger1.redirect('error', logger2);
```
[the remark this answers is not in the document]
Comment (atian25) [reply test/lib/logger.test.js]: ok

**Pull request merged**